### PR TITLE
docs: make sure users anable the strict mode (for zod)

### PR DESCRIPTION
**Pull request by ahhshm**
related to #1506

### CODE REVIEW

File: www/docs/nextjs/nextjs.md [hunk 1]
```diff
@@ -49,9 +49,37 @@ yarn add @trpc/client @trpc/server @trpc/react @trpc/next zod react-query
 ```
 
 - React Query: `@trpc/react` provides a thin wrapper over [react-query](https://react-query.tanstack.com/overview). It is required as a peer dependency.
-- Zod: most examples use Zod for input validation, though it isn't required. You can use a validation library of your choice (Yup, [Superstruct](https://github.com/ianstormtaylor/superstruct), io-ts, etc). In fact, any object containing a `parse`, `create` or `validateSync` method will work.
+- Zod: most examples use [Zod](https://github.com/colinhacks/zod) for input validation and we highly recommended it, though it isn't required. You can use a validation library of your choice ([Yup](https://github.com/jquense/yup), [Superstruct](https://github.com/ianstormtaylor/superstruct), [io-ts](https://github.com/gcanti/io-ts), etc). In fact, any object containing a `parse`, `create` or `validateSync` method will work.
 
-### 2. Create a tRPC router
+### 2. Enable the strict mode
+
+If you want to use Zod for input validation, make sure you have enabled the strict mode in your `tsconfig.js`:
```
Comment (sachinraja) [inline www/docs/nextjs/nextjs.md:56]:
```suggestion
If you want to use Zod for input validation, make sure you have enabled the strict mode in your `tsconfig.json`:
```

File: www/docs/nextjs/nextjs.md [hunk 2]
```diff
@@ -49,9 +49,37 @@ yarn add @trpc/client @trpc/server @trpc/react @trpc/next zod react-query
 ```
 
 - React Query: `@trpc/react` provides a thin wrapper over [react-query](https://react-query.tanstack.com/overview). It is required as a peer dependency.
-- Zod: most examples use Zod for input validation, though it isn't required. You can use a validation library of your choice (Yup, [Superstruct](https://github.com/ianstormtaylor/superstruct), io-ts, etc). In fact, any object containing a `parse`, `create` or `validateSync` method will work.
+- Zod: most examples use [Zod](https://github.com/colinhacks/zod) for input validation and we highly recommended it, though it isn't required. You can use a validation library of your choice ([Yup](https://github.com/jquense/yup), [Superstruct](https://github.com/ianstormtaylor/superstruct), [io-ts](https://github.com/gcanti/io-ts), etc). In fact, any object containing a `parse`, `create` or `validateSync` method will work.
 
-### 2. Create a tRPC router
+### 2. Enable the strict mode
```
Comment (sachinraja) [inline www/docs/nextjs/nextjs.md:54]:
```suggestion
### 2. Enable strict mode
```

File: www/docs/nextjs/nextjs.md [hunk 3]
```diff
@@ -49,9 +49,37 @@ yarn add @trpc/client @trpc/server @trpc/react @trpc/next zod react-query
...
+{
+  // ...
+  "compilerOptions": {
+    // ...
+    "strict": true
+  }
+}
+```
+
+If the strict mode is too much, at least enable `strictNullChecks`:
```
Comment (sachinraja) [inline www/docs/nextjs/nextjs.md:69]:
```suggestion
If strict mode is too much, at least enable `strictNullChecks`:
```

**Review by sachinraja — DISMISSED** [review]
Looks good, just left a few minor suggestions. I'm not sure if it's possible, but it would be nice to be able to duplicate this info across guides without copying it across files. [Maybe we should use MDX files?](https://docusaurus.io/docs/next/markdown-features/react#importing-markdown)

File: www/docs/react/react-intro.md [hunk 4]
```diff
@@ -6,9 +6,10 @@ slug: /react
 ---
 
 :::info
+
 - If you're using Next.js, read the [Usage with Next.js](/docs/nextjs) guide instead.
 - In order to infer types from your Node.js backend you should have the frontend & backend in the same monorepo.
-:::
+  :::
```
Comment (sachinraja) [inline www/docs/react/react-intro.md:12]:
```suggestion
:::
```

**Review by TheoBr — APPROVED** [review]
This would help a lot w/ the less technical people I help set up on tRPC 🙏

**Review by KATT — APPROVED** [review]

**Pull request merged**